### PR TITLE
Add `Credential` to dev config for AWS

### PR DESCRIPTION
**Pull request by wahabmk**
Adding `Credential` to the YAML files for AWS under `dev/config`.  This is needed after https://github.com/Mirantis/hmc/pull/342 was merged. Without this the HMC Operator keeps on failing with:
```sh
ster", "ManagedCluster": {"name":"wali-aws-dev","namespace":"hmc-system"}, "namespace": "hmc-system", "name": "wali-aws-dev", "reconcileID": "de457532-16dd-466a-afe0-e8c5d410ecf4", "error": "Credential.hmc.mirantis.com \"\" not found"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:316
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:263
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.19.0/pkg/internal/controller/controller.go:224
```

### CODE REVIEW

File: config/dev/aws-credentials.yaml [hunk 1]
```diff
@@ -19,4 +19,16 @@ type: Opaque
 stringData:
   AccessKeyID: ${AWS_ACCESS_KEY_ID}
   SecretAccessKey: ${AWS_SECRET_ACCESS_KEY}
-  SessionToken: ${AWS_SESSION_TOKEN}
\ No newline at end of file
+  SessionToken: ${AWS_SESSION_TOKEN}
+---
+apiVersion: hmc.mirantis.com/v1alpha1
+kind: Credential
```
Comment (squizzi) [inline config/dev/aws-credentials.yaml:25]: Do the other providers need this too?

Comment (wahabmk) [reply config/dev/aws-credentials.yaml:25]: Yes, but I want to get this in for AWS at least so it can make my dev cycle simpler 😄

Comment (squizzi) [reply config/dev/aws-credentials.yaml:25]: fwiw, they all share the same format here: `credential` in `ManagedCluster.config` and a `Credential` Kind with `identityRef` so it should be a relatively simple add.

**Comment by wahabmk** [comment]
@a13x5 done

File: config/dev/azure-credentials.yaml [hunk 2]
```diff
@@ -23,3 +23,15 @@ metadata:
...
 type: Opaque
+---
+apiVersion: hmc.mirantis.com/v1alpha1
+kind: Credential
+metadata:
+  name: azure-cluster-identity-cred
+  namespace: ${NAMESPACE}
+spec:
+  identityRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
```
Comment (a13x5) [inline config/dev/azure-credentials.yaml:34]: Wrong API version. Should be `infrastructure.cluster.x-k8s.io/v1beta1`

File: config/dev/vsphere-credentials.yaml [hunk 3]
```diff
@@ -18,3 +18,15 @@ metadata:
...
   password: ${VSPHERE_PASSWORD}
+---
+apiVersion: hmc.mirantis.com/v1alpha1
+kind: Credential
+metadata:
+  name: vsphere-cluster-identity-cred
+  namespace: ${NAMESPACE}
+spec:
+  identityRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
```
Comment (a13x5) [inline config/dev/vsphere-credentials.yaml:29]: Same here - wrong version.

**Review by a13x5 — CHANGES_REQUESTED** [review]
API versions are incorrect, rest looks good

File: config/dev/azure-credentials.yaml (hunk 2 above)
Comment (wahabmk) [reply config/dev/azure-credentials.yaml]: Fixed, thanks!

**Review by squizzi — APPROVED** [review]

**Review by a13x5 — APPROVED** [review]
LGTM

**Pull request merged**